internal/k8s/controller.go

```diff
@@ -910,6 +915,60 @@ func (lbc *LoadBalancerController) syncEndpointSlices(task task) bool {
 	return resourcesFound
 }
 
+// finds the number of currently active endpoints for the service pointing at the ingresscontroller and updates all configs that depend on that number
+func (lbc *LoadBalancerController) updateNumberOfIngressControllerReplicas(controllerEndpointSlice discovery_v1.EndpointSlice) bool {
+	previous := lbc.configurator.GetIngressControllerReplicas()
+	current := countReadyEndpoints(controllerEndpointSlice)
+	found := false
+
+	if current != previous {
+		// number of active endpoints changed. Update configuration of all ingresses that depend on it
+		lbc.configurator.SetIngressControllerReplicas(len(controllerEndpointSlice.Endpoints))
+
+		// handle ingresses
+		resources := lbc.configuration.FindIngressesWithRatelimitScaling(controllerEndpointSlice.Namespace)
+		resourceExes := lbc.createExtendedResources(resources)
+		for _, ingress := range resourceExes.IngressExes {
+			found = true
+			_, err := lbc.configurator.AddOrUpdateIngress(ingress)
+			if err != nil {
+				glog.Errorf("Error updating ratelimit for Ingress %s/%s: %s", ingress.Ingress.Namespace, ingress.Ingress.Name, err)
+			}
+		}
+		for _, ingress := range resourceExes.MergeableIngresses {
+			found = true
+			_, err := lbc.configurator.AddOrUpdateMergeableIngress(ingress)
+			if err != nil {
+				glog.Errorf("Error updating ratelimit for Ingress %s/%s: %s", ingress.Master.Ingress.Namespace, ingress.Master.Ingress.Name, err)
+			}
+		}
+
+		// handle virtualservers
```

Comment:
@dbaumgarten something we observed in regression, the code below was causing panic when CRD flag was not set so we've wrapped it in

```
if lbc.areCustomResourcesEnabled {}
```

Reply:
Oh. Nice find.
Yes, wrapping it in that if sounds like a viable solution 👍